src/nft/components/profile/list/NFTListingsGrid.tsx

```diff
@@ -0,0 +1,550 @@
+  }, [globalPrice])
+
+  let warning: ListingWarning | undefined = undefined
+  if (asset.listingWarnings && asset.listingWarnings?.length > 0) {
+    if (showMarketplaceLogo) {
+      for (const listingWarning of asset.listingWarnings) {
+        if (listingWarning.marketplace.name === selectedMarkets[0].name) warning = listingWarning
+      }
+    } else {
+      warning = asset.listingWarnings[0]
+    }
+  }
+
+  return (
+    <Row transition="500" marginLeft={selectedMarkets.length > 1 ? '20' : '0'}>
+      {showMarketplaceLogo && (
+        <Column
+          position="relative"
+          cursor="pointer"
+          onMouseEnter={handleHover}
+          onMouseLeave={handleHover}
+          style={{ marginLeft: '-28px' }}
+          onClick={(e) => {
+            e.stopPropagation()
+            removeAssetMarketplace(asset, selectedMarkets[0])
+            removeMarket && removeMarket()
+          }}
+        >
+          <Box className={styles.removeMarketplace} visibility={hovered ? 'visible' : 'hidden'} position="absolute">
+            <Box as="img" width="32" src="/nft/svgs/minusCircle.svg" alt="Remove item" />
+          </Box>
+          <Box
+            as="img"
+            alt={selectedMarkets[0].name}
+            width="28"
+            height="28"
+            borderRadius="4"
+            objectFit="cover"
+            src={selectedMarkets[0].icon}
+            marginRight="16"
+          />
+        </Column>
+      )}
+      <Column style={{ flex: '1.5' }}>
+        {globalPriceMethod === SetPriceMethod.SAME_PRICE && !globalOverride ? (
+          <PriceTextInput
+            listPrice={globalPrice}
+            setListPrice={setGlobalPrice}
+            isGlobalPrice={true}
+            setGlobalOverride={setGlobalOverride}
+            globalOverride={globalOverride}
+            warning={warning}
+            asset={asset}
+          />
+        ) : (
+          <PriceTextInput
+            listPrice={listPrice}
+            setListPrice={setListPrice}
+            isGlobalPrice={false}
+            setGlobalOverride={setGlobalOverride}
+            globalOverride={globalOverride}
+            warning={warning}
+            asset={asset}
+          />
+        )}
+      </Column>
+      <Row flex="1" display={{ sm: 'none', md: 'flex' }}>
+        <Box className={body} color="textSecondary" width="full" textAlign="right">
+          {marketplaceFee > 0 ? marketplaceFee + (selectedMarkets.length > 1 ? '% MAX' : '%') : '--%'}
+        </Box>
+      </Row>
+      <Row flex="1" display={{ sm: 'none', md: 'flex' }}>
+        <Box className={body} color="textSecondary" width="full" textAlign="right">
+          {(asset.basisPoints * 0.01).toFixed(1)}%
+        </Box>
+      </Row>
+      <Row style={{ flex: '1.5' }} display={{ sm: 'none', md: 'flex' }}>
+        <Column width="full">
+          <EthPriceDisplay ethPrice={userReceives} />
+          {(showGlobalPrice ? globalPrice && globalPrice !== 0 : listPrice !== 0) && (
```

Comment:
could you make this a variable? ideally we would do if blocks bc its sorta hard to follow. some comments also would be helpful here